lib/cfg-parser.c

```diff
@@ -462,17 +462,14 @@ cfg_parser_cleanup(CfgParser *self, gpointer instance)
 gboolean
 cfg_process_flag(CfgFlagHandler *handlers, gpointer base, const gchar *flag_)
 {
-  gint h;
-  gchar flag[32];
+  gsize h;
 
-  for (h = 0; flag_[h] && h < sizeof(flag); h++)
+  GString *flag = g_string_new(flag_);
```

Comment:
Make sure to use flag->str in the strcmp below.

Also please free up the new GString.

Reply:
Thanks!